portfolio/src/main/java/com/google/sps/servlets/NewCommentServlet.java

```diff
@@ -27,27 +46,158 @@
-
+    taskEntity.setProperty("blobKey", blobKey);
+    taskEntity.setProperty("landmarkName", landmarkName);
+    taskEntity.setProperty("landmarkGeoPt", landmarkGeoPt);
+    
     datastore.put(taskEntity);
 
     response.sendRedirect("/pages/server-dev.html");
   }
+
+  /**
+   * Returns a BlobKey object corresponding to the uploaded file.
+   *
+   * @param request The <code>HttpServletRequest</code> for the POST request.
+   * @param formInputElementName The name attribute of the image file input to the form.
+   * @return The blob key associated with the uploaded image file. Null is returned if the user did
+   *     not select a file or the file is not an image type.
+   */
+  private BlobKey getBlobKey(HttpServletRequest request, String formInputElementName) {
+    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+    List<BlobKey> blobKeys = blobs.get(formInputElementName);
+
+    // User submitted form without selecting a file, so we can't get a URL. (dev server)
+    if (blobKeys == null || blobKeys.isEmpty()) {
+      return null;
+    }
+
+    // Our form only contains a single file input, so get the first index.
+    BlobKey blobKey = blobKeys.get(0);
+
+    // User submitted form without selecting a file, so we can't get a URL. (live server)
+    BlobInfo blobInfo = new BlobInfoFactory().loadBlobInfo(blobKey);
+    if (blobInfo.getSize() == 0) {
+      blobstoreService.delete(blobKey);
+      return null;
+    }
+
+    // Check the validity of the file here by making sure it's an image file
+    if (!"image".equals(blobInfo.getContentType().substring(0, 5))) {
+      blobstoreService.delete(blobKey);
+      return null;
+    }
+
+    return blobKey;
+  }
+
+  /**
+   * Blobstore stores files as binary data. This function retrieves the binary data stored at the
+   * BlobKey parameter.
+   *
+   * @param blobKey The key associated with the image whose binary data is retrieved.
+   * @return An byte array containing the binary data of the image associated with the blobKey.
+   * @throws IOException - If an output error occurs when writing bytes from the temp blobstore 
+   *                       buffer <code>b</code> to the output byte array <code>outputBytes</code>.
+   */
+  private byte[] getBlobBytes(BlobKey blobKey) throws IOException {
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    ByteArrayOutputStream outputBytes = new ByteArrayOutputStream();
+
+    int fetchSize = BlobstoreService.MAX_BLOB_FETCH_SIZE;
+    long currentByteIndex = 0;
+    boolean continueReading = true;
+    while (continueReading) {
+      // end index is inclusive, so we have to subtract 1 to get fetchSize bytes
+      byte[] b =
+          blobstoreService.fetchData(blobKey, currentByteIndex, currentByteIndex + fetchSize - 1);
+      outputBytes.write(b);
+
+      // if we read fewer bytes than we requested, then we reached the end
+      if (b.length < fetchSize) {
```

Comment:
do you want to update this fetchSize somewhere? otherwise, why not just using MAX_BLOB_FETCH_SIZE?

Reply:
Addressed in #28.